lib/template-loader.php

```diff
@@ -81,24 +81,44 @@ function gutenberg_override_query_template( $template, $type, array $templates =
  * @param array $block The root block to start traversing from.
  */
 function create_auto_draft_for_template_part_block( $block ) {
-	if ( 'core/template-part' === $block['blockName'] && ! isset( $block['attrs']['id'] ) ) {
-		$template_part_file_path =
-			get_stylesheet_directory() . '/block-template-parts/' . $block['attrs']['slug'] . '.html';
-		if ( ! file_exists( $template_part_file_path ) ) {
-			return;
+	global $_wp_current_template_part_ids;
+
+	if ( 'core/template-part' === $block['blockName'] ) {
+		if ( ! isset( $block['attrs']['postId'] ) ) {
+			$template_part_file_path =
```

Comment:
Do you think it would be worthwhile to extract this to `get_template_part_file_path( $slug )` helper? It might help with readability here due to nesting, and I imagine we might need this resolution logic elsewhere. It can be abstracted further to include both templates and template parts if needed, since it seems that they follow the same pattern?

Reply:
The demo block template code is temporary until we have a proper block-based theme; this will flatten once we can remove it.